### PR TITLE
Improve the connection flow dropdown

### PR DESCRIPTION
**Pull request by improbable-valentyn**
#### Description
I combined https://github.com/spatialos/UnrealGDK/pull/2290 with the Connection Flow dropdown improvements suggested in https://improbableio.atlassian.net/browse/UNR-3746:
- `Cloud Deployment Name` field in the dropdown now refers to the same property as `Deployment Name` in the Cloud Deployment Configuration window, so the `Start Deployment` toolbar button will now use the name specified in the dropdown when quickly starting the new deployment without going through the Cloud Deployment Configuration window.
- `Local Deployment IP` and `Cloud Deployment Name` labels now get grayed out correctly when the edit box is disabled.
- Entering an invalid IP into the `Exposed local runtime IP address` field in the editor settings will trigger a warning popup and reset the value to an empty string.

#### Primary reviewers
@jessicafalk @MatthewSandfordImprobable

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp [hunk 1]
```diff
@@ -1010,13 +1031,13 @@ void FSpatialGDKEditorToolbarModule::CloudDeploymentClicked()
...
 }
 
-bool FSpatialGDKEditorToolbarModule::IsLocalDeploymentIPEditable() const
+bool FSpatialGDKEditorToolbarModule::IsLocalDeploymentIPEditable()
 {
 	const USpatialGDKEditorSettings* SpatialGDKEditorSettings = GetDefault<USpatialGDKEditorSettings>();
 	return GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking() && (SpatialGDKEditorSettings->SpatialOSNetFlowType == ESpatialOSNetFlow::LocalDeployment);
 }
 
-bool FSpatialGDKEditorToolbarModule::AreCloudDeploymentPropertiesEditable() const
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp:1019]: No longer const?

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp:1019]: Ahh static :p

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Private/SpatialGDKEditorToolbar.cpp:1019]: It's static now.

File: SpatialGDK/Source/SpatialGDKEditorToolbar/Public/SpatialGDKEditorToolbar.h [hunk 2]
```diff
@@ -109,8 +110,8 @@ class FSpatialGDKEditorToolbarModule : public IModuleInterface, public FTickable
 	void LocalDeploymentClicked();
 	void CloudDeploymentClicked();
 
-	bool IsLocalDeploymentIPEditable() const;
-	bool AreCloudDeploymentPropertiesEditable() const;
+	static bool IsLocalDeploymentIPEditable();
```
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDKEditorToolbar/Public/SpatialGDKEditorToolbar.h:113]: Is there a standard here where we make member function static if they can be?

If so I guess GetPrimaryDeploymentNameText can be static?

Comment (improbable-valentyn) [reply SpatialGDK/Source/SpatialGDKEditorToolbar/Public/SpatialGDKEditorToolbar.h:113]: Yes, I also noticed a lot of the functions in this class as well as `SSpatialGDKCloudDeploymentConfiguration` can be static, but I felt it wasn't really necessary to change them in this PR and we don't really gain much by doing that.

**Pull request merged**